migrate/migrator.go

```diff
@@ -47,6 +47,15 @@ func WithTemplateData(data any) MigratorOption {
 	}
 }
 
+// WithSort overrides the default ascending sort function for all migrations.
+// This affects all sorting operations in the entire migrate package.
+func WithSort(ascSortFn, descSortFn MigrationSortFunc) {
```

Comment:
```suggestion
func SetSort(ascSortFn, descSortFn MigrationSortFunc) {
```

nit: can we rename this to `SetSort` so as to make it visually distinct from MigratorOptions/MigrationOptions above?

We use this naming pattern for global options in other places too, e.g.:
https://github.com/uptrace/bun/blob/e135221c1942e6e69fd0be67b7dfaf072ecc61d4/schema/table.go#L34